storage/oasis/nodeapi/api.go

```diff
@@ -16,13 +16,12 @@ import (
 	consensus "github.com/oasisprotocol/nexus/coreapi/v22.2.11/consensus/api"
 	consensusTransaction "github.com/oasisprotocol/nexus/coreapi/v22.2.11/consensus/api/transaction"
 	consensusResults "github.com/oasisprotocol/nexus/coreapi/v22.2.11/consensus/api/transaction/results"
-	genesis "github.com/oasisprotocol/nexus/coreapi/v22.2.11/genesis/api"
-	governance "github.com/oasisprotocol/nexus/coreapi/v22.2.11/governance/api"
 	registry "github.com/oasisprotocol/nexus/coreapi/v22.2.11/registry/api"
 	roothash "github.com/oasisprotocol/nexus/coreapi/v22.2.11/roothash/api"
 	"github.com/oasisprotocol/nexus/coreapi/v22.2.11/roothash/api/message"
 	scheduler "github.com/oasisprotocol/nexus/coreapi/v22.2.11/scheduler/api"
 	staking "github.com/oasisprotocol/nexus/coreapi/v22.2.11/staking/api"
+	governance "github.com/oasisprotocol/nexus/coreapi/v24.0/governance/api"
```

Comment:
Mind adding a doc-level comment explaining how these types work? Perhaps something similar to the one in `compat_types.go`:

``` The nodeapi package provides a low-level interface to the Oasis node API.
Due to the (sometimes breaking) changes in the types used by different 
evolutions of oasis-core over the Cobalt, Damask, and Eden upgrades, 
Nexus defines its own set of internal types here. 

The types that it exposes are mostly simplified versions of the types exposed by
oasis-core Damask: The top-level type structs are defined in this file, and
the types of their fields are almost universally directly the types exposed
by oasis-core Damask. The reason is that as oasis-core evolves, Damask types
are mostly able to represent all the information from Cobalt, plus some. However, 
there are a few exceptions where we use oasis-core Eden types instead, e.g. in the 
`governance` module.

When a new type is added, devs should try to convert it to the oasis-core Damask 
type first. If that is not possible, devs can either define a custom internal type or
repackage the latest (Eden) type as the Nexus-internal type. The first option is 
mainly used when we want to pare down a given type to only retain the fields 
Nexus needs.
```

Feel free to reword; that last paragraph may be unnecessary.

Reply:
Thanks, mainly added what you wrote. I skipped the last paragraph and added a sentence that over time, the types should be migrated to newer versions as needed.